File: src/main/java/com/ballgame/Rectangle.java
package com.ballgame;

public abstract class Rectangle extends GameObject {
    protected int width;
    protected int height;

    public Rectangle(int x, int y) {
        super(x, y);
    }

    public double getCenterX() {
        return this.x + (this.width / 2);
    }

    public double getCenterY() {
        return this.y + (this.height / 2);
    }

    public int getWidth() {
        return this.width;
    }

    public int getHeight() {
        return this.height;
    }
}

File: src/main/kotlin/com/ballgame/Rectangle.kt
package com.ballgame

abstract class Rectangle(x: Int, y: Int) : GameObject(x, y) {
	var width = 0
		protected set
	var height = 0
		protected set
	val centerX = x + (width / 2)
	val centerY = y + (height / 2)
}
